src/Microsoft.TemplateEngine.Utils/EngineEnvironmentSettings.cs

```diff
@@ -57,7 +59,7 @@ public string BaseDir
                 {
                     if (_baseDir == null)
                     {
-                        _baseDir = Path.Combine(UserProfileDir, ".netnew");
+                        _baseDir = Path.Combine(UserProfileDir, ".netnew", Host.AssemblyName, Host.AssemblyVersion.ToString());
```

Comment:
This one should get captured in the renaming/changing of the name/version pair in the host settings